app/src/main/java/me/magnum/melonds/ui/layouteditor/LayoutEditorView.kt

```diff
@@ -168,11 +169,12 @@ class LayoutEditorView(context: Context, attrs: AttributeSet?) : LayoutView(cont
     private fun dragView(view: View, offsetX: Float, offsetY: Float) {
         val finalX = min(max(view.x + offsetX, 0f), width - view.width.toFloat())
         val finalY = min(max(view.y + offsetY, 0f), height - view.height.toFloat())
-        val newParams = LayoutParams(view.width, view.height).apply {
-            leftMargin = finalX.toInt()
-            topMargin = finalY.toInt()
+        view.updateLayoutParams {
```

Comment:
This usage of `updateLayoutParams` is also not working correctly. But instead of just fixing this, and I know that this is not related to the changes you have made, an improvement can be made here. Instead of modifying the layout params of the view here directly, we can delegate that to the `LayoutComponentView` to keep the logic encapsulated. This can be accomplished by receiving a `LayoutComponentView` as the first parameter (instead of a `View`) and then call `setPosition(Point)` to update the view's position. For example, something like this:

```kt
private fun dragView(view: LayoutComponentView, offsetX: Float, offsetY: Float) {
    val currentPosition = view.getPosition()
    val finalX = min(max(currentPosition.x + offsetX, 0f), width - view.getWidth().toFloat())
    val finalY = min(max(currentPosition.y + offsetY, 0f), height - view.getHeight().toFloat())
    view.setPosition(Point(finalX.toInt(), finalY.toInt()))
}
```

Then, you just need to modify the place that calls this method, where the `LayoutComponentView` instance is already available.

Reply:
I see... 🤔